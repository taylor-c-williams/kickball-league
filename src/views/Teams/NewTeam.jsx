import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { createTeam } from '../../services/teams';
import TeamNewForm from '../../components/Teams/TeamNewForm';

export default function NewTeam() {
	const [name, setName] = useState('');
	const [city, setCity] = useState('');
	const [state, setState] = useState('');
	const history = useHistory();

	const handleSubmit = async (e) => {
		e.preventDefault();
		const res = await createTeam({ name, city, state });
		history.push(`/teams/${res[0].id}`);
	};
	return (
		<div>
			<h1>Create New Team</h1>
			<TeamNewForm
				name={name}
				city={city}
				state={state}
				handleSubmit={handleSubmit}
				setName={setName}
				setCity={setCity}
				setState={setState}
			/>
		</div>
	);
}
