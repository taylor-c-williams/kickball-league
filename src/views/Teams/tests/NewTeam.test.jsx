import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import NewTeam from '../NewTeam';

it('should render accurately the create team view and component', async () => {
	const container = render(
		<MemoryRouter initialEntries={['/newTeam']}>
			<Route path="/newTeam">
				<NewTeam />
			</Route>
		</MemoryRouter>
	);

	const textEntry = await screen.findByText('Create stuff', {
		exact: false,
	});

	expect(container).toMatchSnapshot();
	expect(textEntry).toBeInTheDocument();
});
