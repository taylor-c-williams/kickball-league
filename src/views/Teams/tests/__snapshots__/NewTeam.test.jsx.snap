// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render accurately the create team view and component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Create New Team
        </h1>
        <div>
          <fieldset>
            <legend>
              Details
            </legend>
            <form>
              <label
                for="name"
              >
                Team Name:
              </label>
              <input
                id="name"
                name="name"
                type="text"
                value=""
              />
              <label
                for="city"
              >
                City:
              </label>
              <input
                id="city"
                name="city"
                type="text"
                value=""
              />
              <label
                for="state"
              >
                State:
              </label>
              <input
                id="state"
                name="state"
                type="text"
                value=""
              />
              <button
                aria-label="Add New Team"
                type="submit"
                value="Add New Team"
              >
                Add New Team
              </button>
            </form>
            <section
              class="title-img"
            >
              <img
                alt="kickball"
                height="400"
                src="/kickballs/ball_4.jpg"
              />
            </section>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Create New Team
      </h1>
      <div>
        <fieldset>
          <legend>
            Details
          </legend>
          <form>
            <label
              for="name"
            >
              Team Name:
            </label>
            <input
              id="name"
              name="name"
              type="text"
              value=""
            />
            <label
              for="city"
            >
              City:
            </label>
            <input
              id="city"
              name="city"
              type="text"
              value=""
            />
            <label
              for="state"
            >
              State:
            </label>
            <input
              id="state"
              name="state"
              type="text"
              value=""
            />
            <button
              aria-label="Add New Team"
              type="submit"
              value="Add New Team"
            >
              Add New Team
            </button>
          </form>
          <section
            class="title-img"
          >
            <img
              alt="kickball"
              height="400"
              src="/kickballs/ball_4.jpg"
            />
          </section>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
